docs/source/model_design/programming_vivarium/debugging/index.rst

```diff
@@ -0,0 +1,190 @@
+..
+  Section title decorators for this document:
+  
+  ==============
+  Document Title
+  ==============
+  Section Level 1
+  ---------------
+  Section Level 2
+  +++++++++++++++
+  Section Level 3
+  ~~~~~~~~~~~~~~~
+  Section Level 4
+  ^^^^^^^^^^^^^^^
+  Section Level 5
+  '''''''''''''''
+
+  The depth of each section level is determined by the order in which each
+  decorator is encountered below. If you need an even deeper section level, just
+  choose a new decorator symbol from the list here:
+  https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections
+  And then add it to the list of decorators above.
+
+.. _debugging_rt:
+
+===============
+Basic Debugging
+===============
+
+.. contents::
+   :local:
+   :depth: 2
+
+Debugging is an art and science that we couldn't hope to cover on this 
+single page. Rather than attempt such foolishness, we we review a few simple 
+techniques in the context of a targeted example here. If you find the 
+bug to be outside the scope of this page, it's likely time to ask an 
+engineer for help.  
+
+Making Sense of the Stack Trace
+-------------------------------
+
+The stack trace is an output that the model provides to show what 
+error message was received and trace back the flow of code to what 
+caused the error. Often the immediate line of code that generated the error will be 
+significantly removed from the line of code that will need to be changed 
+to fix the error. 
+
+Let's work through an example to understand this. Say a line of code 
+we edit in the :code:`loader.py` file filters a dataset incorrectly 
```

Comment:
Also, I had no idea what `loader.py` was (I still don't, really, since I've never had to build an Artifact), and I didn't know where to look for it. So I looked at the other pages in this section and eventually found it mentioned on the Artifact Building page, but that page also didn't make it clear where the function came from, so I had to poke around until I found it in the `vivarium_research_template` repo [here](https://github.com/ihmeuw/vivarium_research_template/blob/main/%7B%7Bcookiecutter.package_name%7D%7D/src/%7B%7Bcookiecutter.package_name%7D%7D/data/loader.py). So before jumping straight into the example, I think you should provide a bit of context, like "Suppose you're building an Artifact, as described on [[link to Artifact Buliding page]]," and when you mention the `loader.py` file, it would be helpful to link to the actual file so the reader can see what sorts of things are in there.

Reply:
Sure - I also linked to the overview of engineering files page which is designed to help explain the different files in the engineering repos. https://vivarium-research.readthedocs.io/en/latest/model_design/programming_vivarium/overview_engineering_files/index.html